centreon/src/Core/Host/Infrastructure/Repository/DbReadHostRepository.php

```diff
@@ -134,6 +136,48 @@ public function exists(int $hostId): bool
         return (bool) $statement->fetchColumn();
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function existsByAccessGroups(int $hostId, array $accessGroups): bool
+    {
+        if (empty($accessGroups)) {
+            $this->debug('Access groups array empty');
+
+            return false;
+        }
+
+        $accessGroupIds = array_map(
+            static fn (AccessGroup $accessGroup): int => $accessGroup->getId(),
+            $accessGroups
+        );
+
+        $concatenator = new SqlConcatenator();
+
+        $concatenator->defineSelect(
+            <<<'SQL'
+                SELECT 1
+                FROM `:db`.host h
+                JOIN `:dbstg`.centreon_acl acl
+                    ON h.host_id = acl.host_id
```

Comment:
There's no need to join the host table, you can check directly with the centreon_acl table.